lib/portus/ldap.rb

```diff
@@ -90,11 +90,16 @@ def bind_options
       search_options.merge(password: password)
     end
 
+    # Returns the LDAP attribute where to search for username.
+    def uid_attribute
+      APP_CONFIG["ldap"]["uid"] || "uid"
+    end
+
     # Returns the hash to be used in order to search for a user in the LDAP
     # server.
     def search_options
       {}.tap do |opts|
-        opts[:filter] = "(uid=#{username})"
+        opts[:filter] = "(#{uid_attribute}=#{username})"
```

Comment:
You can just use `APP_CONFIG["ldap"]["uid"]`. This is because the `config.yml` is guaranteed to have the default values. If a user wants to override this, it has to either write it on the `config-local.yml` file, or use the proper environment variable.